homework/index.js

```diff
@@ -30,15 +32,138 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-        return;
-      }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+  function openNewTab(url) {
+    window.open(url, '_blank');
+  }
+
+  function bindContributions(contributorsUrl) {
+    const repoInfoContainer = document.getElementById('repo-info-container');
+    const contributorsDiv = createAndAppend('div', repoInfoContainer, {
+      class: 'contributors-info',
+    });
+    fetchJSON(contributorsUrl)
+      .then(response => {
+        const div = createAndAppend('div', contributorsDiv);
+        createAndAppend('h4', div, { text: 'Contributions' });
+        createAndAppend('hr', div);
+
+        const table = createAndAppend('ul', div, { class: 'grid' });
+        const contributors = response;
+        contributors.forEach((_contributor, index) => {
+          const imageUrl = contributors[index].avatar_url;
+          const name = contributors[index].login;
+          const contributionsCount = contributors[index].contributions;
+          const profileURL = contributors[index].html_url;
+          const tr = createAndAppend('li', table, { class: 'clickable row' });
+          tr.onclick = () => openNewTab(profileURL);
+          const td1 = createAndAppend('div', tr, { class: 'col-30' });
+          const imgContainerDiv = createAndAppend('div', td1, {
+            class: 'image-container',
+          });
+          createAndAppend('img', imgContainerDiv, {
+            src: imageUrl,
+            width: 80,
+            height: 80,
+          });
+          createAndAppend('div', tr, {
+            text: name,
+            class: 'col-30 description',
+          });
+          const td3 = createAndAppend('div', tr, { class: 'col-30' });
+          createAndAppend('label', td3, { text: contributionsCount });
+        });
+      })
+      .catch(error => {
+        createAndAppend('div', contributorsDiv, {
+          text: error.message,
+          class: 'alert-error',
+        });
+      });
+  }
+  function bindRepositoryDetails(selectedRepository) {
+    const { description, forks } = selectedRepository;
+    const repositoryName = selectedRepository.name;
```

Comment:
Since you are now using object destructuring in line 84, you as well in the `name` property and use that instead of the `repositoryName` const:

```js
function bindRepositoryDetails(selectedRepository) {
  const { name, description, forks } = selectedRepository;
  ...
  createAndAppend('a', repoNameCol, {
    href: repoURL,
    target: '_blank',
    text: name, // <- here
  });
```